### DISCUSSION TITLE
SF5.4 onAuthenticationSuccess without PHPSESSID cookies ?

### DISCUSSION THREAD

**Discussion by thgauthier** [symfony/symfony#44611] [Q&A]
Hello,
I try to work with a custom authenticator.
I don't understand why  onAuthenticationSuccess is called and there's no  PHPSESSID cookies.
As a result after RedirectResponse the auth process start again in an infinite loop...
![security yaml](https://user-images.githubusercontent.com/95478233/145986762-248aa931-5ba3-4144-a6ba-a5b5ac7f8f44.png)
![AppCustomAuthenticator php](https://user-images.githubusercontent.com/95478233/145987206-0d48e44a-fe03-44f2-b4aa-89b8e2acad3d.png)
![result-Mozilla Firefox](https://user-images.githubusercontent.com/95478233/145988053-0c88e652-e6fa-4a1a-91cc-b0255a20def2.png)

**Comment by wouterj** [comment]
What you're observing is correct, "the auth process start again in an infinite loop" is not good.

`onAuthenticationSuccess()` is called before persisting the token in the session - this only happens at the very end of the request. So you should not see the token in the session.

My guess would be that something causes deauthentication upon loading the next page. Maybe your user object changes (e.g. you edit the roles) or something else. You can use the web dev profiler "Logs" and "Security" tabs of the next page to find out what causes deauthentication.

**Reply by thgauthier** [reply]
Thank you very much for your answer.
Even with a simple `return new Response('onAuthenticationSuccess !!!');` in the onAuthenticationSuccess() method the PHPSESSID id change each time I reload the page. Once one authentication successfully, I expect to reach my home controller but it never happen and I can't see the web dev profiler...
I can't see what what could be wrong in var/log/dev.log
Any idea ?

![1](https://user-images.githubusercontent.com/95478233/146026617-49d27ce6-4281-4461-9684-11b5a582f78c.jpg)
![2](https://user-images.githubusercontent.com/95478233/146026658-e3fa9565-62e7-426c-95da-01fdab628762.jpg)
![3](https://user-images.githubusercontent.com/95478233/146026672-91c93b78-b652-47e5-9445-4763e970d894.jpg)
![4](https://user-images.githubusercontent.com/95478233/146026689-d00bca7e-22d3-4319-9234-686cb0a1d80e.jpg)

<pre>
xxxxxxx@xxxx:~/xxxxxxx[bugfix/remote_user !?]$ cat var/log/dev.log

#0 /home/xxxxxxx/xxxxxxx/var/cache/dev/ContainerY0251qn/App_KernelDevDebugContainer.php(542): ContainerY0251qn\\getSession_Storage_NativeService::do()
#1 /home/xxxxxxx/xxxxxxx/vendor/symfony/dependency-injection/Container.php(422): ContainerY0251qn\\App_KernelDevDebugContainer->load()
#2 /home/xxxxxxx/xxxxxxx/vendor/symfony/dependency-injection/Argument/ServiceLocator.php(42): Symfony\\Component\\DependencyInjection\\Container->getService()
#3 /home/xxxxxxx/xxxxxxx/vendor/symfony/http-kernel/EventListener/SessionListener.php(40): Symfony\\Component\\DependencyInjection\\Argument\\ServiceLocator->get()
#4 /home/xxxxxxx/xxxxxxx/vendor/symfony/event-dispatcher/Debug/WrappedListener.php(117): Symfony\\Component\\HttpKernel\\EventListener\\SessionListener->onKernelRequest()
#5 /home/xxxxxxx/xxxxxxx/vendor/symfony/event-dispatcher/EventDispatcher.php(230): Symfony\\Component\\EventDispatcher\\Debug\\WrappedListener->__invoke()
#6 /home/xxxxxxx/xxxxxxx/vendor/symfony/event-dispatcher/EventDispatcher.php(59): Symfony\\Component\\EventDispatcher\\EventDispatcher->callListeners()
#7 /home/xxxxxxx/xxxxxxx/vendor/symfony/event-dispatcher/Debug/TraceableEventDispatcher.php(154): Symfony\\Component\\EventDispatcher\\EventDispatcher->dispatch()
#8 /home/xxxxxxx/xxxxxxx/vendor/symfony/http-kernel/HttpKernel.php(128): Symfony\\Component\\EventDispatcher\\Debug\\TraceableEventDispatcher->dispatch()
#9 /home/xxxxxxx/xxxxxxx/vendor/symfony/http-kernel/HttpKernel.php(74): Symfony\\Component\\HttpKernel\\HttpKernel->handleRaw()
#10 /home/xxxxxxx/xxxxxxx/vendor/symfony/http-kernel/Kernel.php(202): Symfony\\Component\\HttpKernel\\HttpKernel->handle()
#11 /home/xxxxxxx/xxxxxxx/public/index.php(20): Symfony\\Component\\HttpKernel\\Kernel->handle()
#12 {main}
"} []

#0 /home/xxxxxxx/xxxxxxx/var/cache/dev/ContainerY0251qn/App_KernelDevDebugContainer.php(542): ContainerY0251qn\\getSession_Storage_MetadataBagService::do()
#1 /home/xxxxxxx/xxxxxxx/var/cache/dev/ContainerY0251qn/getSession_Storage_NativeService.php(24): ContainerY0251qn\\App_KernelDevDebugContainer->load()
#2 /home/xxxxxxx/xxxxxxx/var/cache/dev/ContainerY0251qn/App_KernelDevDebugContainer.php(542): ContainerY0251qn\\getSession_Storage_NativeService::do()
#3 /home/xxxxxxx/xxxxxxx/vendor/symfony/dependency-injection/Container.php(422): ContainerY0251qn\\App_KernelDevDebugContainer->load()
#4 /home/xxxxxxx/xxxxxxx/vendor/symfony/dependency-injection/Argument/ServiceLocator.php(42): Symfony\\Component\\DependencyInjection\\Container->getService()
#5 /home/xxxxxxx/xxxxxxx/vendor/symfony/http-kernel/EventListener/SessionListener.php(40): Symfony\\Component\\DependencyInjection\\Argument\\ServiceLocator->get()
#6 /home/xxxxxxx/xxxxxxx/vendor/symfony/event-dispatcher/Debug/WrappedListener.php(117): Symfony\\Component\\HttpKernel\\EventListener\\SessionListener->onKernelRequest()
#7 /home/xxxxxxx/xxxxxxx/vendor/symfony/event-dispatcher/EventDispatcher.php(230): Symfony\\Component\\EventDispatcher\\Debug\\WrappedListener->__invoke()
#8 /home/xxxxxxx/xxxxxxx/vendor/symfony/event-dispatcher/EventDispatcher.php(59): Symfony\\Component\\EventDispatcher\\EventDispatcher->callListeners()
#9 /home/xxxxxxx/xxxxxxx/vendor/symfony/event-dispatcher/Debug/TraceableEventDispatcher.php(154): Symfony\\Component\\EventDispatcher\\EventDispatcher->dispatch()
#10 /home/xxxxxxx/xxxxxxx/vendor/symfony/http-kernel/HttpKernel.php(128): Symfony\\Component\\EventDispatcher\\Debug\\TraceableEventDispatcher->dispatch()
#11 /home/xxxxxxx/xxxxxxx/vendor/symfony/http-kernel/HttpKernel.php(74): Symfony\\Component\\HttpKernel\\HttpKernel->handleRaw()
#12 /home/xxxxxxx/xxxxxxx/vendor/symfony/http-kernel/Kernel.php(202): Symfony\\Component\\HttpKernel\\HttpKernel->handle()
#13 /home/xxxxxxx/xxxxxxx/public/index.php(20): Symfony\\Component\\HttpKernel\\Kernel->handle()
#14 {main}
"} []
[2021-12-14T14:42:32.953019+00:00] request.INFO: Matched route "home". {"route":"home","route_parameters":{"_route":"home","_controller":"App\\Controller\\HomeController::index"},"request_uri":"https://xxx.xxx.xxx/~xxxxxxx/","method":"GET"} []
[2021-12-14T14:42:32.960905+00:00] security.DEBUG: Checking for authenticator support. {"firewall_name":"main","authenticators":1} []
[2021-12-14T14:42:32.960962+00:00] security.DEBUG: Checking support on authenticator. {"firewall_name":"main","authenticator":"App\\Security\\AppCustomAuthenticator"} []
[2021-12-14T14:42:32.962412+00:00] php.INFO: User Deprecated: Since symfony/framework-bundle 5.3: The "session.storage.factory.service" service is deprecated, use "session.storage.factory.native", "session.storage.factory.php_bridge" or "session.storage.factory.mock_file" instead. {"exception":"[object] (ErrorException(code: 0): User Deprecated: Since symfony/framework-bundle 5.3: The \"session.storage.factory.service\" service is deprecated, use \"session.storage.factory.native\", \"session.storage.factory.php_bridge\" or \"session.storage.factory.mock_file\" instead. at /home/xxxxxxx/xxxxxxx/var/cache/dev/ContainerY0251qn/getSession_Storage_Factory_ServiceService.php:22)
[stacktrace]
#0 /home/xxxxxxx/xxxxxxx/var/cache/dev/ContainerY0251qn/App_KernelDevDebugContainer.php(542): ContainerY0251qn\\getSession_Storage_Factory_ServiceService::do()
#1 /home/xxxxxxx/xxxxxxx/var/cache/dev/ContainerY0251qn/getSession_FactoryService.php(23): ContainerY0251qn\\App_KernelDevDebugContainer->load()
#2 /home/xxxxxxx/xxxxxxx/var/cache/dev/ContainerY0251qn/App_KernelDevDebugContainer.php(542): ContainerY0251qn\\getSession_FactoryService::do()
#3 /home/xxxxxxx/xxxxxxx/var/cache/dev/ContainerY0251qn/get_Container_Private_SessionService.php(23): ContainerY0251qn\\App_KernelDevDebugContainer->load()
#4 /home/xxxxxxx/xxxxxxx/var/cache/dev/ContainerY0251qn/App_KernelDevDebugContainer.php(542): ContainerY0251qn\\get_Container_Private_SessionService::do()
#5 /home/xxxxxxx/xxxxxxx/vendor/symfony/dependency-injection/Container.php(422): ContainerY0251qn\\App_KernelDevDebugContainer->load()
#6 /home/xxxxxxx/xxxxxxx/vendor/symfony/dependency-injection/Argument/ServiceLocator.php(42): Symfony\\Component\\DependencyInjection\\Container->getService()
#7 /home/xxxxxxx/xxxxxxx/vendor/symfony/http-kernel/EventListener/SessionListener.php(51): Symfony\\Component\\DependencyInjection\\Argument\\ServiceLocator->get()
#8 /home/xxxxxxx/xxxxxxx/vendor/symfony/http-kernel/EventListener/AbstractSessionListener.php(74): Symfony\\Component\\HttpKernel\\EventListener\\SessionListener->getSession()
#9 /home/xxxxxxx/xxxxxxx/vendor/symfony/http-foundation/Request.php(735): Symfony\\Component\\HttpKernel\\EventListener\\AbstractSessionListener->Symfony\\Component\\HttpKernel\\EventListener\\{closure}()
#10 /home/xxxxxxx/xxxxxxx/vendor/symfony/http-foundation/Request.php(754): Symfony\\Component\\HttpFoundation\\Request->getSession()
#11 /home/xxxxxxx/xxxxxxx/vendor/symfony/security-http/Firewall/ContextListener.php(98): Symfony\\Component\\HttpFoundation\\Request->hasPreviousSession()
#12 /home/xxxxxxx/xxxxxxx/vendor/symfony/security-bundle/Debug/WrappedLazyListener.php(49): Symfony\\Component\\Security\\Http\\Firewall\\ContextListener->authenticate()
#13 /home/xxxxxxx/xxxxxxx/vendor/symfony/security-bundle/Security/LazyFirewallContext.php(60): Symfony\\Bundle\\SecurityBundle\\Debug\\WrappedLazyListener->authenticate()
#14 /home/xxxxxxx/xxxxxxx/vendor/symfony/security-bundle/Debug/TraceableFirewallListener.php(70): Symfony\\Bundle\\SecurityBundle\\Security\\LazyFirewallContext->__invoke()
#15 /home/xxxxxxx/xxxxxxx/vendor/symfony/security-http/Firewall.php(92): Symfony\\Bundle\\SecurityBundle\\Debug\\TraceableFirewallListener->callListeners()
#16 /home/xxxxxxx/xxxxxxx/vendor/symfony/event-dispatcher/Debug/WrappedListener.php(117): Symfony\\Component\\Security\\Http\\Firewall->onKernelRequest()
#17 /home/xxxxxxx/xxxxxxx/vendor/symfony/event-dispatcher/EventDispatcher.php(230): Symfony\\Component\\EventDispatcher\\Debug\\WrappedListener->__invoke()
#18 /home/xxxxxxx/xxxxxxx/vendor/symfony/event-dispatcher/EventDispatcher.php(59): Symfony\\Component\\EventDispatcher\\EventDispatcher->callListeners()
#19 /home/xxxxxxx/xxxxxxx/vendor/symfony/event-dispatcher/Debug/TraceableEventDispatcher.php(154): Symfony\\Component\\EventDispatcher\\EventDispatcher->dispatch()
#20 /home/xxxxxxx/xxxxxxx/vendor/symfony/http-kernel/HttpKernel.php(128): Symfony\\Component\\EventDispatcher\\Debug\\TraceableEventDispatcher->dispatch()
#21 /home/xxxxxxx/xxxxxxx/vendor/symfony/http-kernel/HttpKernel.php(74): Symfony\\Component\\HttpKernel\\HttpKernel->handleRaw()
#22 /home/xxxxxxx/xxxxxxx/vendor/symfony/http-kernel/Kernel.php(202): Symfony\\Component\\HttpKernel\\HttpKernel->handle()
#23 /home/xxxxxxx/xxxxxxx/public/index.php(20): Symfony\\Component\\HttpKernel\\Kernel->handle()
#24 {main}
"} []
[2021-12-14T14:42:32.972856+00:00] security.DEBUG: Read existing security token from the session. {"key":"_security_main","token_class":"Symfony\\Component\\Security\\Http\\Authenticator\\Token\\PostAuthenticationToken"} []
[2021-12-14T14:42:33.018428+00:00] doctrine.DEBUG: SELECT 
[2021-12-14T14:42:33.027301+00:00] doctrine.DEBUG: SELECT 
[2021-12-14T14:42:33.032964+00:00] doctrine.DEBUG: SELECT 
[2021-12-14T14:42:33.033351+00:00] security.DEBUG: User was reloaded from a user provider. {"provider":"Symfony\\Bridge\\Doctrine\\Security\\User\\EntityUserProvider","username":"toto@toto.com"} []
[2021-12-14T14:42:33.037127+00:00] doctrine.DEBUG: SELECT 
[2021-12-14T14:42:33.038675+00:00] security.INFO: Authenticator successful! {"token":{"Symfony\\Component\\Security\\Http\\Authenticator\\Token\\PostAuthenticationToken":"PostAuthenticationToken(user=\"toto@toto.com\", authenticated=true, roles=\"ROLE_SUP_ADM, ROLE_ADMINISTRATOR, ROLE_DUPLICATE_EVENT_CUSTOM_QUERIES\")"},"authenticator":"Symfony\\Component\\Security\\Http\\Authenticator\\Debug\\TraceableAuthenticator"} []
[2021-12-14T14:42:33.045315+00:00] security.DEBUG: The "Symfony\Component\Security\Http\Authenticator\Debug\TraceableAuthenticator" authenticator set the response. Any later authenticator will not be called {"authenticator":"Symfony\\Component\\Security\\Http\\Authenticator\\Debug\\TraceableAuthenticator"} []
[2021-12-14T14:42:33.062251+00:00] security.DEBUG: Stored the security token in the session. {"key":"_security_main"} []
</pre>

**Comment by berkut1** [comment]
SF 5.4.1 has problem with session. And in some scenario also in 5.4.0
You can find temporarily fixes there https://github.com/symfony/symfony/issues/44541

**Comment by kevincerro** [comment]
> Hello, I try to work with a custom authenticator. I don't understand why onAuthenticationSuccess is called and there's no PHPSESSID cookies. As a result after RedirectResponse the auth process start again in an infinite loop... ![security yaml](https://user-images.githubusercontent.com/95478233/145986762-248aa931-5ba3-4144-a6ba-a5b5ac7f8f44.png) ![AppCustomAuthenticator php](https://user-images.githubusercontent.com/95478233/145987206-0d48e44a-fe03-44f2-b4aa-89b8e2acad3d.png) ![result-Mozilla Firefox](https://user-images.githubusercontent.com/95478233/145988053-0c88e652-e6fa-4a1a-91cc-b0255a20def2.png)

I have the same issue

Symfony 5.4.2 and PHP 7.4.13 & 8.1
